src/entity/User.ts

```diff
@@ -101,22 +101,24 @@ export class User {
 
 export type AddUserData = Pick<
   User,
-  'id' | 'name' | 'image' | 'username' | 'email' | 'createdAt' | 'github'
+  | 'id'
+  | 'name'
+  | 'image'
+  | 'username'
+  | 'email'
+  | 'createdAt'
+  | 'github'
+  | 'referral'
+  | 'infoConfirmed'
+  | 'profileConfirmed'
 >;
 export type UpdateUserEmailData = Pick<User, 'id' | 'email'>;
 type AddNewUserResult =
   | { status: 'ok'; userId: string }
   | { status: 'failed'; reason: UserFailErrorKeys; error?: Error };
 
 const checkRequiredFields = (data: AddUserData): boolean => {
-  return !!(
-    data &&
-    data.id &&
-    data.name &&
-    data.image &&
-    data.username &&
-    data.email
-  );
+  return !!(data && data.id && data.name && data.image);
```

Comment:
`image` and `name` can also potentially be empty

Reply:
Image should always have a fallback set as far as I can see.
Name is true, i'll remove that.

Reply:
Could be, but we can't rely on that. It's better to allow null and set the fallback before return using GQL resolvers or something just in case. What do you think?